### ISSUE TITLE
Error "process exited with signal SIGSEGV" in Javascript

### ISSUE THREAD

**Issue by juusechec** [LeetCode-Feedback/LeetCode-Feedback#7230]
#### Your LeetCode username
naturalmentejorge

#### Category of the bug
- [X] Question
- [ ] Solution
- [ ] Language
- [ ] Missing Test Cases 

#### Description of the bug

I'm gettirng "process exited with signal SIGSEGV" error in javascript with a simple algorithm. I am responding the test https://leetcode.com/contest/weekly-contest-292/problems/count-nodes-equal-to-average-of-subtree/

#### Code you used for Submit/Run operation

```js
function getLevel(index, lastLevel) {
  const reindex = index + 1;
  const relevel = lastLevel + 1;
  if (reindex < 2 ** relevel) {
    return lastLevel;
  } else {
    return lastLevel + 1;
  }
}

function createTree(arr, indexarr, level) {
  level = getLevel(indexarr, level);
  if (indexarr >= arr.length) {
    return null;
  }
  var node = {};
  node.val = arr[indexarr];
  var indexLeft = ((indexarr + 1) * 2) - 1;
  var indexRight = indexLeft + 1;
  node.left = createTree(arr, indexLeft, level);
  node.right = createTree(arr, indexRight, level);
  return node;
}

var averageOfSubtree = function (root) {
  var treeRoot = createTree(root, 0, 0);
  console.log("treeRoot:", treeRoot);
  return "to be completed";
};

console.log(averageOfSubtree([4,8,5,0,1,null,6]));
console.log(averageOfSubtree([1]));
```

#### Language used for code
Javascript

#### Expected behavior

The expected print is this (because it is the result on my local nodejs):
```
treeRoot: {
  val: 4,
  left: {
    val: 8,
    left: { val: 0, left: null, right: null },
    right: { val: 1, left: null, right: null }
  },
  right: {
    val: 5,
    left: { val: null, left: null, right: null },
    right: { val: 6, left: null, right: null }
  }
}

treeRoot: { val: 1, left: null, right: null }
```

The script uses approximately 2.79 MB

#### Screenshots

![image](https://user-images.githubusercontent.com/4140058/167286317-95a73a7d-a8e8-4007-af9e-0753216b2b80.png)

#### Additional context

I'm measuring the memory usage on localhost with:
```javascript
const used = process.memoryUsage().heapUsed / 1024 / 1024;
console.log(`The script uses approximately ${Math.round(used * 100) / 100} MB`);
```

**Comment by LC-Yani** [comment]
Thank you for reaching out to us. I've relayed this issue to our team to investigate.

**Comment by LC-Pam** [comment]
Thank you for reaching out to us, the sample test case is correct and the judger appears to be working correctly, please feel free to refer to the discuss section of the problem to see how other users' have optimized their codes!

**Comment by timfenney** [comment]
@LC-Pam I don't think this should be closed, as the solution is written in Javascript.
Please tell the team it is Javascript and not C++ or something. _Javascript_ on its own shouldn't be able to segfault, as we can't specify memory addresses in the first place.
The error is likely something else in the process. For example, if you are running Javascript solutions with Node, there is either a bug in Node, or it was built incorrectly, or there is a misbehaving native extension. But for sure it is not the JS code submitted in this solution.

**Comment by danicruz0415** [comment]
This error message is totally unhelpful and is not an error that JavaScript would throw. Its true that the code may be faulty but the system should at least pass the original error that the JS engine had, instead of this _probably_ wrongly handled exception. I discovered that this issue (at least in my case) was caused by an stack overflow. The original error thrown by the JS engine was "Uncaught RangeError: Maximum call stack size exceeded". I had to "submit" for the platform to show me the faulty test case because when you hit "Run" it does not even show you what test case failed.
